src/modals/AnonymizedDataConsentScreen.tsx

```diff
@@ -26,39 +20,44 @@ const AnonymizedDataConsentScreen: FunctionComponent = () => {
     healthAuthorityName,
     healthAuthorityPrivacyPolicyUrl,
   } = useConfigurationContext()
-  const { updateUserConsent } = useAnalyticsContext()
+  const { userConsentedToAnalytics, updateUserConsent } = useAnalyticsContext()
 
-  const handleOnPressConsent = async () => {
-    await updateUserConsent(true)
+  const handleOnPressButton = async () => {
+    const nextConsentState = !userConsentedToAnalytics
+    updateUserConsent(nextConsentState)
     navigation.navigate(Stacks.Settings)
   }
 
   const handleOnPressClose = () => {
-    navigation.navigate(Stacks.Settings)
+    navigation.goBack()
   }
 
+  const headerText = userConsentedToAnalytics
+    ? t("settings.you_are_sharing_data")
+    : t("settings.share_anonymized_data")
+
+  const buttonText = userConsentedToAnalytics
+    ? t("settings.stop_sharing_data")
+    : t("settings.i_understand_and_consent")
+
   return (
-    <>
-      <StatusBar backgroundColor={Colors.primaryLightBackground} />
-      <ScrollView
-        style={style.container}
-        contentContainerStyle={style.mainContentContainer}
-      >
+    <View style={style.container}>
```

Comment:
Does this need a `ScrollView`?

Reply:
I don't think so! Currently the outer View is not scrollable, and the header (that says "You are sharing data" and has the close button) is fixed. The view that contains the information about data sharing and the CTA button is scrollable